pydantic/main.py

```diff
@@ -804,3 +734,181 @@ def validate_model(  # noqa: C901 (ignore complexity)
+    allowed_keys: Optional['SetStr'] = None,
+    skip_defaults: bool = False,
+    as_type: Optional[Type['BaseModel']] = None,
+) -> 'TupleGenerator':
+    self_type = type(model)
+    for k, v in model.__dict__.items():
+        if allowed_keys is None or k in allowed_keys:
+            casting_type = None
+            if as_type is not None:
+                target_field = as_type.__fields__[k]
+                if _requires_casting(v, target_field, self_type.__fields__[k]):
+                    casting_type = target_field.type_
+            yield k, _get_value_fast(
+                v,
+                to_dict=to_dict,
+                by_alias=by_alias,
+                skip_defaults=skip_defaults,
+                as_type=casting_type,
+            )
+
+
+@no_type_check
+def _get_value(
+    v: Any,
+    to_dict: bool,
+    by_alias: bool,
+    include: Optional[Union['SetIntStr', 'DictIntStrAny']],
+    exclude: Optional[Union['SetIntStr', 'DictIntStrAny']],
+    skip_defaults: bool,
+    as_type: Optional[Type[Any]] = None,
+) -> Any:
+    if isinstance(v, BaseModel):
+        if to_dict:
+            return v.dict(
+                by_alias=by_alias, skip_defaults=skip_defaults, include=include, exclude=exclude, as_type=as_type
+            )
+        else:
+            return v.copy(include=include, exclude=exclude)
+    if not (include or exclude):
+        return _get_value_fast(
+            v, to_dict, by_alias, skip_defaults, as_type
+        )
+
+    value_exclude = ValueItems(v, exclude) if exclude else None
+    value_include = ValueItems(v, include) if include else None
+
+    if isinstance(v, dict):
+        return {
+            k_: _get_value(
+                v_,
+                to_dict=to_dict,
+                by_alias=by_alias,
+                skip_defaults=skip_defaults,
+                include=value_include and value_include.for_element(k_),
+                exclude=value_exclude and value_exclude.for_element(k_),
+                as_type=as_type,
+            )
+            for k_, v_ in v.items()
+            if (not value_exclude or not value_exclude.is_excluded(k_))
+               and (not value_include or value_include.is_included(k_))
+        }
+
+    elif isinstance(v, (list, set, tuple)):
+        return type(v)(
+            _get_value(
+                v_,
+                to_dict=to_dict,
+                by_alias=by_alias,
+                skip_defaults=skip_defaults,
+                include=value_include and value_include.for_element(i),
+                exclude=value_exclude and value_exclude.for_element(i),
+                as_type=as_type,
+            )
+            for i, v_ in enumerate(v)
+            if (not value_exclude or not value_exclude.is_excluded(i))
+            and (not value_include or value_include.is_included(i))
+        )
+
+    else:
+        if as_type is not None:
+            return parse_as_type(v, as_type)
+        return v
+
+
+@no_type_check
+def _get_value_fast(
```

Comment:
This function is *substantially* faster than the `_get_value` function due to the lack of checks related to include/exclude. (The same goes for `_iter_fast`, to a lesser degree.)